openassessment/xblock/static/js/src/lms/oa_staff_area.js

```diff
@@ -148,117 +279,222 @@
+                return enabled;
+            }
+        },
+
+        /**
+         * Called when something is selected or typed in the assessment rubric.
+         * Used to set the unsaved changes warning dialog.
+         *
+         * @param {string} key the unsaved changes key
+         * @param {boolean} changesExist true if unsaved changes exist
+         */
+        assessmentRubricChanges: function(key, changesExist) {
+            if (changesExist) {
+                this.baseView.unsavedWarningEnabled(
+                    true,
+                    key,
+                    gettext("If you leave this page without submitting your staff assessment, you will lose any work you have done.") // jscs:ignore maximumLineLength
+                );
+            }
+        },
+
+        /**
+         * Submit the staff assessment override.
+         *
+         * @param {string} submissionID The ID of the submission to be submitted.
+         * @param {element} rubric The rubric element to be assessed.
+         * @param {element} scope An ancestor element for the submit button (to allow for shared
+         *     classes in different form).
+         */
+        submitStaffOverride: function(submissionID, rubric, scope) {
+            var view = this;
+            var successCallback = function() {
+                view.baseView.unsavedWarningEnabled(false, view.OVERRIDE_UNSAVED_WARNING_KEY);
+                // Note: we ignore any message returned from the server and instead
+                // re-render the student info with the "Learner's Final Grade"
+                // section expanded. This section will show the learner's
+                // final grade and in the future should include details of
+                // the staff override itself.
+                view.loadStudentInfo({expanded_view: 'final-grade'});
+            };
+            this.callStaffAssess(submissionID, rubric, scope, successCallback, '.staff-override-error');
+        },
+
+        /**
+         * Submit the staff grade, and check out another learner for grading if continueGrading is true.
+         *
+         * @param {string} submissionID The ID of the submission to be submitted.
+         * @param {element} rubric The rubric element to be assessed.
+         * @param {element} scope An ancestor element for the submit button (to allow for shared
+         *     classes in different form).
+         * @param {boolean} continueGrading If true, another learner will be marked as "In Progress",
+         *     and a new grading form will be rendered with the learner's answer.
+         */
+        submitStaffGrade: function(submissionID, rubric, scope, continueGrading) {
+            var view = this;
+            var successCallback = function() {
+                view.baseView.unsavedWarningEnabled(false, view.FULL_GRADE_UNSAVED_WARNING_KEY);
+                view.staffGradeFormLoaded = false;
+                view.loadStaffGradeForm(!continueGrading);
```

Comment:
@cptvitamin, I realized after setting up the sandbox that we need to scroll up in the "continueGrading" case or else the staff member can't tell that a new form has been loaded for grading. I have added that scroll operation in #827. I still wonder what we need to do with focus though.